src/nomad_simulations/properties/energies.py

```diff
@@ -77,3 +78,502 @@ def __init__(
+    calculated using the method described in XC_functional.
+    """
+
+    # ! Someone check this description!
+    # ? Do we really want to specify the method here? This can't be user-defined?
+
+    value = Quantity(
+        type=np.float64,
+        unit='joule',
+        description="""
+        The value of the correlation energy.
+        """,
+    )
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+# ? XCExchangeEnergy?
+class ExchangeEnergy(QuantumEnergy):
+    """
+    Physical property section describing the exchange energy of a (sub)system,
+    calculated using the method described in XC_functional.
+    """
+    # ! Someone check this description!
+    # ? Do we really want to specify the method here? This can't be user-defined?
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+class ZeroTemperatureEnergy(QuantumEnergy):
+    """
+    Physical property section describing the total energy of a (sub)system extrapolated to $T=0$, based on a free-electron gas argument.
+    """
+    # ! Someone check this description!
+    # ? Do we really want to specify the method here? This can't be user-defined?
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+class ZeroPointEnergy(QuantumEnergy):
+    """
+    Physical property section describing the zero-point vibrational energy of a (sub)system,
+    calculated using the method described in zero_point_method.
+    """
+    # ! Someone check this description!
+    # ? Do we really want to specify the method here? This can't be user-defined?
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+class ElectrostaticEnergy(QuantumEnergy):
+    """
+    Physical property section describing the electrostatic energy (nuclei + electrons) of a (sub)system.
+    """
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+class NuclearRepulsionEnergy(QuantumEnergy):
+    """
+    Physical property section describing the nuclear-nuclear repulsion energy of a (sub)system.
+    """
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+##########################
+# Other / General energies
+##########################
+
+
+
+    # madelung = SubSection(
+    #     sub_section=EnergyEntry.m_def,
+    #     description="""
+    #     Contains the value and information regarding the Madelung energy.
+    #     """,
+    # )
+
+    # # TODO I suggest ewald is moved to "long range" under electrostatic->energyentry, unless there is some other usage I am misunderstanding
+    # ewald = SubSection(
+    #     sub_section=EnergyEntry.m_def,
+    #     description="""
+    #     Contains the value and information regarding the Ewald energy.
+    #     """,
+    # )
```

Comment:
Subtype of electrostatic energy.

Reply:
This is a major question that I have: How should we deal with sub-contributions? One option is to keep a flat list (flat inside total_energy.contributions at least), and somehow distinguish sub contributions with names and descriptions.

The other option is to nest. If we do this, I think we should be restrictive about the nesting, i.e., specify energy sub-contributions that are relevant for a particular contribution. We should not allow free recursive nesting or sub-contributions that don't make sense theoretically/methodologically

Reply:
Or you can have both options at once if you use inheritance.